src/lib/crudAppointment.js

```diff
@@ -0,0 +1,195 @@
+import { Appointment } from "./models/appointment.js";
+
+// Function to open the modal to be inserted in the + button (new appointment)
+export function openModal() {
+  const modal = document.getElementById("myModal");
+  modal.style.display = "block";
+}
+
+// Function to close the modal when clicking outside the modal content
+window.onclick = function (event) {
+  const modal = document.getElementById("myModal");
+  if (event.target == modal) {
+    modal.style.display = "none";
+  }
+};
+
+// Generate a unique ID
+function generateId() {
+  return Math.floor(Math.random() * 999999);
+}
+
+// User-generated data extracted from HTML page
+const DESCRIPTION_KEY = "description";
+const DUEDATE_KEY = "dueDate"; // appointment due date
+const CATEGORY_KEY = "category";
+const USERID_KEY = "userId";
+
+// Retrieve data from HTML form
+export function getDataFromForm() {
+  const description = document.getElementById(DESCRIPTION_KEY).value;
+  const dueDate = new Date(document.getElementById(DUEDATE_KEY).value);
+  const category = document.getElementById(CATEGORY_KEY).value;
+
+  // Retrieve user ID from localStorage
+  const userId = localStorage.getItem(USERID_KEY);
+
+  // Generate appointment ID
+  const id = generateId();
+
+  // Create creation date
+  const creationDate = new Date();
+
+  // Initially, updateDate and completionDate are null
+  const updateDate = null;
+  const completionDate = null;
+
+  // Create appointment
+  createAppointment(
+    id,
+    userId,
+    description,
+    creationDate,
+    updateDate,
+    completionDate,
+    dueDate,
+    category
+  );
+}
+
+// Event listener to call getDataFromForm when the form is submitted
+document
+  .getElementById("yourFormId") // INSERT CORRECT HTML ID !!!!!!!!!!!!!!!!!!
+  .addEventListener("submit", function (event) {
+    event.preventDefault();
+    getDataFromForm();
+  });
+
+// Create appointment
+export function createAppointment(
+  id,
+  userId,
+  description,
+  creationDate,
+  updateDate,
+  completionDate,
+  dueDate,
+  category
+) {
+  const appointment = new Appointment(
+    id,
+    userId,
+    description,
+    creationDate,
+    updateDate,
+    completionDate,
+    dueDate,
+    category
+  );
+  localStorage.setItem(appointment.id, JSON.stringify(appointment));
```

Comment:
We don't want to store the single appointment by id in the localStorage,
we want to store an array of appointment instead, so, in this function you should get the old list from the localStorage, update the list, and save it.
Something like this:
const appointments = JSON.parse(localStorage.get(APPOINTMENTS_KEY))
appointments.push(newAppointments)
localStorage.set(APPOINTMENTS_KEY, JSON.stringify(appointments))